### PR TITLE
Proxy ping/pong keepalive

### PR DESCRIPTION
**Pull request by stafyniaksacha**
Add ping / pong system to keep alive connection between kuzzle proxy and kuzzle server

## New .kuzzlerc options :
```javascript
{
  "proxyBroker": {
     "host": "localhost",
     "port": 7331,
     "retryInterval": 1000,
     "pingTimeout": 50, // Timeout between ping request and pong response
     "pingInterval": 60000 // Interval between a successful pong response and another ping request
   }
}
```

## Boyscout

* Upgrade Elasticsearch minor version from 5.0 to 5.1
* Fixed collection & index information in forbidden error message (funnel controller)
* Fixed document garbage collector, context was lost during 2nd run (kuiz crash)

### CODE REVIEW

File: .kuzzlerc.sample [hunk 1]
```diff
@@ -283,7 +283,9 @@
       "aliases": ["broker"],
       "host": "localhost",
       "port": 7911,
-      "retryInterval": 1000
+      "retryInterval": 1000,
+      "pingTimeout": 50,
```
Comment (scottinet) [inline .kuzzlerc.sample]: These two new options must be documented in the sample file, using the comments immediately above the `internalBroker` and `proxyBroker` configuration sections.

File: lib/services/garbageCollector.js [hunk 2]
```diff
@@ -40,13 +40,13 @@ function GarbageCollector(kuzzle) {
 
     if (kuzzle.funnel.overloaded) {
       // Reschedule the task to run in 1 hour
-      this.timer = setTimeout(this.run, OneHour);
+      this.timer = setTimeout(this.run.bind(this), OneHour);
```
Comment (scottinet) [inline lib/services/garbageCollector.js]: Since run is a property of the `GarbageCollector` object, I'm not sure binding the context is necessary.

Comment (scottinet) [inline lib/services/garbageCollector.js]: After our discussion: I stand corrected. :-)

File: .kuzzlerc.sample (hunk 1 above)
Comment (stafyniaksacha) [inline .kuzzlerc.sample]: Done

**Pull request merged**